scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -115,6 +119,58 @@ const pasteAddSuggestion = (state, {content, data}) => {
     return onChange(state, editorState, true);
 };
 
+/**
+ * @ngdoc method
+ * @name processSuggestion
+ * @param {Object} state
+ * @param {Object} selection - the selection of suggestion
+ * @param {Boolean} accepted - the suggestion is accepted
+ * @return {Object} returns new state
+ * @description Accept or reject the suggestions in the selection.
+ */
+const processSuggestion = (state, {selection}, accepted) => {
+    const types = ['DELETE_SUGGESTION', 'ADD_SUGGESTION'];
+    const start = selection.getStartOffset();
+    const end = selection.getEndOffset();
+    let {editorState} = state;
+    let content = editorState.getCurrentContent();
+    let entityKey;
+    let entity;
+    let type;
+
+    for (let i = end - start - 1; i >= 0; i--) {
+        entityKey = getEntityKey(content, getEntityKeyByOffset(content, selection, i), types);
+        if (entityKey == null) {
+            continue;
+        }
+
+        entity = content.getEntity(entityKey);
+        if (entity == null) {
+            continue;
+        }
+        type = entity.get('type');
+
+        const applySuggestion = type === 'ADD_SUGGESTION' && accepted || type === 'DELETE_SUGGESTION' && !accepted;
+        const newSelection = selection.merge({
+            anchorOffset: start + i + (applySuggestion ? 0 : 1),
+            focusOffset: start + i + (applySuggestion ? 0 : 1),
+            isBackward: false
+        });
+
+        editorState = EditorState.acceptSelection(editorState, newSelection);
```

Comment:
maybe you should use `forceSelection` here to fix the popup closing, but not sure

Reply:
After I rebased it is working fine.